OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -63,6 +63,8 @@ public class ArmamentInfo : ConditionalTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		public readonly bool ShootAtClosestTargetablePosition = false;
```

Comment:
Can we tackle this from the other direction?  Set up the new behaviour as the default, but then have different Targetable traits that expose the additional targetable locations vs only the center location.  This was how I originally designed / intented our targeting to work, before #8727 collapsed everything into the one trait.